src/main/scala/nl.knaw.dans.easy.stage/dataset/Remarks.scala

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (C) 2015 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.stage.dataset
+
+import java.io.{ File, FileNotFoundException }
+import java.nio.charset.StandardCharsets
+import java.nio.file.{ Files, NoSuchFileException, Paths }
+
+import nl.knaw.dans.lib.error._
+import nl.knaw.dans.lib.logging.DebugEnhancedLogging
+import nl.knaw.dans.lib.string._
+import org.apache.commons.lang.BooleanUtils
+
+import scala.util.Try
+import scala.xml.{ Elem, XML }
+
+case class Remarks(bagitDir: File) extends DebugEnhancedLogging {
```

Comment:
given the structure of the class, would it make sense to do the following:

```scala
case class Agreements(xml: Node) {
  // acceptedLicense, privacySensitiveRemark and messageFromDepositor here
  // perhaps even as 'lazy val' instead of 'def', given that the answer doesn't change if you ask a second time and no side effects should be performed
}

object Agreements {
  def apply(bagitDir: File): Try[Option[Agreements]] = {
    // construct path to 'agreements.xml'
    // if the file is not present, return 'Success(Option.empty)'
    // else, parse the file into XML (wrapped in 'Try') and return an 'Agreements'
  }
}
```

Reply:
The whole class is created, used and abandoned in a single for comprehension. So why bother premature optimization?